### PR TITLE
Docs: vectara + NLI hallucination metric

### PR DESCRIPTION
**Pull request by marko-polo-cheno**
Fixes KOL-4114

image

### CODE REVIEW

**Comment by marko-polo-cheno** [comment]
Fixes KOL-4112

File: docs/metrics/index.md [hunk 1]
```diff
@@ -127,6 +127,26 @@ biases, and its intended uses.
 
 </div>
 
+## Hallucination
```
Comment (aandyw) [inline docs/metrics/index.md]: I think we want to name this `Large Language Model Evaluation`

File: mkdocs.yml [hunk 2]
```diff
@@ -81,6 +81,8 @@ nav:
       - Computer Vision:
         - metrics/geometry-matching.md
         - metrics/iou.md
+      - Hallucination:
```
Comment (aandyw) [inline mkdocs.yml]: Here as well.

**Review by aandyw — APPROVED** [review]

File: docs/metrics/index.md (hunk 1 above)
Comment (marko-polo-cheno) [reply docs/metrics/index.md]: LLM eval is a subset of NLP, right?

File: mkdocs.yml (hunk 2 above)
Comment (marko-polo-cheno) [reply mkdocs.yml]: 👍

File: docs/metrics/nli-classification.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# NLI Classification
```
Comment (y27choi) [inline docs/metrics/nli-classification.md]: Should we name it by the metric's name?
Are we going to have this doc focused on `contradiction` class so we can identify hallucination?

File: docs/metrics/nli-classification.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+The [Cross-Encoder for Natural Language Inference](https://huggingface.co/cross-encoder/nli-deberta-v3-base) (NLI) is an
```
Comment (y27choi) [inline docs/metrics/nli-classification.md]: `is an` -> `is a`

File: docs/metrics/nli-classification.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+text classification model that takes a pair of text and assigns a label: `'contradiction'`, `'entailment'` or
+`'neutral'`. This is useful for hallucination detection, as factual consistency implies the absence of contradictions.
+For any label, the higher the score, the more confident the model is to assign that label. So, it chooses to assign the
```
Comment (y27choi) [inline docs/metrics/nli-classification.md]: Please add the expected range of the metric.

File: docs/metrics/nli-classification.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+## Implementation Details
+
+This cross-encoder for NLI classification is based on Microsoft's
+[deberta-v3-base](https://huggingface.co/microsoft/deberta-v3-base), which was trained on NLI data, using
+[`sentence_transformers.cross_encoder`](https://www.sbert.net/docs/package_reference/cross_encoder.html).
+
+Further details for development and usage can be found on Hugging Face:
+[cross-encoder/nli-deberta-v3-base](https://huggingface.co/vectara/hallucination_evaluation_model).
+Below is a quick example of how it can be used:
+
```
Comment (y27choi) [inline docs/metrics/nli-classification.md]: Should we have a callout here that walks through the installation steps?

File: docs/metrics/index.md [hunk 7]
```diff
@@ -127,6 +127,26 @@ biases, and its intended uses.
...
+<div class="grid cards" markdown>
+
+- [NLI Classification](nli-classification.md)
+
+    ---
+
+    NLI classification is an insightful tool to measure if candidate texts and reference texts are contradictory or
+    consistent, providing extra detail towards hallucination detection.
+
+- [Vectara](vectara.md)
```
Comment (y27choi) [inline docs/metrics/index.md]: Let's name this as `Vectara's HEM score`

File: docs/metrics/vectara.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+## Implementation Details
+
+HEM is based on Microsoft's [deberta-v3-base](https://huggingface.co/microsoft/deberta-v3-base), which was trained on
+NLI (Natural Language Inference) data, and then fine-tuned on text summarization datasets.
+
+Further details for development and usage can be found on Hugging Face:
+[vectara/hallucination_evaluation_model](https://huggingface.co/vectara/hallucination_evaluation_model).
+Below is a quick example of how it can be used:
+
```
Comment (y27choi) [inline docs/metrics/vectara.md]: Same here.  Let's add a callout for installation steps.

File: docs/metrics/nli-classification.md (hunk 3 above)
Comment (marko-polo-cheno) [reply docs/metrics/nli-classification.md]: Sure, it can just be named contradiction score

File: docs/metrics/nli-classification.md (hunk 6 above)
Comment (marko-polo-cheno) [reply docs/metrics/nli-classification.md]: Adding the necessary pip install, but nothing more. Don't want to go into torch things there

File: docs/metrics/contradiction-score.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+        'neutral': scores[2],
+    }
+
+print(compute_metric("The duck crossed the road", "The duck did not cross the road"))
+# Outputs:
+# {'label': 'contradiction', 'contradiction': 7.436936, 'entailment': -4.0519376, 'neutral': -3.030173}
+```
+
+## Examples
+| Ground Truth | Inference | Classification | Contradiction | Entailment | Neutral |
```
Comment (y27choi) [inline docs/metrics/contradiction-score.md]: One small nit:
Can we add <nobr></nobr> on Classification column?
Example:
`<nobr>Confidence ↓</nobr>`

It moves the last `n` from `contradiction` to the next line.

**Review by y27choi — APPROVED** [review]
Other than one small nit, it looks good. 👍

**Pull request merged**